File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/java/src/lojaexemplo/Produto.java
package lojaexemplo;

public class Produto {

    private int codigo;
    private String nome;
    private double preco;
    private int quantidade;

    public Produto(int codigo, String nome, double preco, int quantidade) {
        this.codigo = codigo;
        this.nome = nome;
        this.preco = preco;
        this.quantidade = quantidade;
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public double getPreco() {
        return preco;
    }

    public void setPreco(double preco) {
        if (preco >= 0) {
            this.preco = preco;
        }
    }

    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        if (quantidade >= 0) {
            this.quantidade = quantidade;
        }
    }

    public double calcularTotal() {
        return preco * quantidade;
    }

    @Override
    public String toString() {
        return "(" + codigo + ") " + nome + " | Preço: " + preco +
                " | Quantidade: " + quantidade +
                " | Total: " + calcularTotal();
    }
}

File: 5_semestre/Programacao-para-Dispositivos-Moveis/UAS/UA1/csharp/LojaExemplo/Produto.cs
namespace LojaExemplo
{
    public class Produto
    {
        public int Codigo { get; set; }
        public string Nome { get; set; }
        public double Preco { get; set; }
        public int Quantidade { get; set; }

        public Produto(int codigo, string nome, double preco, int quantidade)
        {
            Codigo      = codigo;
            Nome        = nome;
            Preco       = preco >= 0 ? preco : 0;
            Quantidade  = quantidade >= 0 ? quantidade : 0;
        }

        public double CalcularTotal()
        {
            return Preco * Quantidade;
        }

        public override string ToString()
        {
            return $"({Codigo}) {Nome}\n" +
                   $"Preço: {Preco}\n" +
                   $"Quantidade: {Quantidade}\n" +
                   $"Total: {CalcularTotal()}";
        }
    }
}